internal/sdkv2provider/resource_cloudflare_waiting_room_settings.go

```diff
@@ -0,0 +1,75 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareWaitingRoomSettings() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceCloudflareWaitingRoomSettingsRead,
+		UpdateContext: resourceCloudflareWaitingRoomSettingsUpdate,
+		CreateContext: resourceCloudflareWaitingRoomSettingsUpdate,
+		DeleteContext: schema.NoopContext,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareWaitingRoomSettingsImport,
+		},
+
+		Schema: resourceCloudflareWaitingRoomSettingsSchema(),
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Second),
+			Update: schema.DefaultTimeout(30 * time.Second),
+		},
+		Description: "Configure zone-wide settings for Cloudflare waiting rooms.",
+	}
+}
+
+func buildWaitingRoomSettingsUpdate(d *schema.ResourceData) cloudflare.UpdateWaitingRoomSettingsParams {
+	searchEngineCrawlerBypass := d.Get("search_engine_crawler_bypass").(bool)
+	return cloudflare.UpdateWaitingRoomSettingsParams{
+		SearchEngineCrawlerBypass: &searchEngineCrawlerBypass,
+	}
+}
+
+func resourceCloudflareWaitingRoomSettingsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Id()
+	if zoneID == "" {
```

Comment:
this shouldn't ever happen; `d.Id()` should always be set in a `Read` operation.